travis/test-appimages.sh

```diff
@@ -14,12 +14,49 @@ error() {
 
 # first of all, try to run appimagetool
 out/appimagetool-"$ARCH".AppImage && error  # should fail due to missing parameter
+out/appimaged-"$ARCH".AppImage && error  # should fail due to missing parameter
 out/appimagetool-"$ARCH".AppImage -h || error  # should not fail
+out/appimaged-"$ARCH".AppImage -h || error  # should not fail
 
-# print version and update information
+# print versions, updateinfo and other info, if present
+# first, appimagetool
 out/appimagetool-"$ARCH".AppImage --version || error  # should not fail
 out/appimagetool-"$ARCH".AppImage --appimage-version || error  # should not fail
 out/appimagetool-"$ARCH".AppImage --appimage-updateinformation || error  # should not fail
+out/appimagetool-"$ARCH".AppImage --appimage-updateinfo || error  # should not fail
+out/appimagetool-"$ARCH".AppImage --appimage-signature || error  # should not fail
+out/appimagetool-"$ARCH".AppImage --appimage-offset || error  # should not fail
+# next, appimaged
+out/appimaged-"$ARCH".AppImage --version || error  # should not fail
+out/appimaged-"$ARCH".AppImage --appimage-version || error  # should not fail
+out/appimaged-"$ARCH".AppImage --appimage-updateinformation || error  # should not fail
+out/appimaged-"$ARCH".AppImage --appimage-updateinfo || error  # should not fail
+out/appimaged-"$ARCH".AppImage --appimage-signature || error  # should not fail
+out/appimaged-"$ARCH".AppImage --appimage-offset || error  # should not fail
+
+# check creation of .home and .config files (even if never used by these tools)
+out/appimagetool-"$ARCH".AppImage --appimage-portable-home || error  # should not fail
+out/appimagetool-"$ARCH".AppImage --appimage-portable-config || error  # should not fail
+out/appimaged-"$ARCH".AppImage --appimage-portable-home || error  # should not fail
+out/appimaged-"$ARCH".AppImage --appimage-portable-config || error  # should not fail
+[ -d out/appimagetool-"$ARCH".AppImage.home ] || error   # should not fail
+[ -d out/appimagetool-"$ARCH".AppImage.config ] || error   # should not fail
+[ -d out/appimaged-"$ARCH".AppImage.home ] || error   # should not fail
+[ -d out/appimaged-"$ARCH".AppImage.config ] || error   # should not fail
+# Don't care cleaning up out/appimage{d,tool}-"$ARCH".AppImage.{config,home}... Or should we?
+
+# now test --appimage-mount
+timeout "$TIMEOUT" out/appimaged-"$ARCH".AppImage    --appimage-mount ; ret_of_appimaged_mount="$?"
+timeout "$TIMEOUT" out/appimagetool-"$ARCH".AppImage --appimage-mount ; ret_of_appimagetool_mount="$?"
+echo ret_of_appimaged_mount=$ret_of_appimaged_mount
```

Comment:
Please remove these `echo` lines. We run the script using `-x` anyway.

Reply:
I added the echo because I wasn't sure if the `-x` would be permanent and persist over time.

Can you tell me: if/where/when is this script called on Travis? With a quick look in the repo I couldn't spot it. Had no time for a thorough grokking of its context...

Reply:
@KurtPfeifle check out `.travis.yml`. It calls `tests/travis-build.sh`, which calls that script.

Regarding the echoes: even if we'd switch that `-x` off, it's a minor change to switch it on again. But if we decide to switch it off, we won't want these extra prints anyway...

Reply:
I see

    Error creating portable home directory at \
      /home/travis/build/AppImage/AppImageKit/build/out/appimagetool-i686.AppImage.home: \
      Permission denied

in https://travis-ci.org/AppImage/AppImageKit/jobs/347849535#L4489

How can that be explained?

If you switch off `-x`, you would still want the final result for each test printed in the log, no? Otherwise, why run the script at all?